lib/httpd_configmap_generator/base/sssd.rb

```diff
@@ -1,58 +1,55 @@
-  class Base
-    def configure_sssd
-      info_msg("Configuring SSSD Service")
-      config = config_file_read(SSSD_CONFIG)
-      configure_sssd_domain(config, domain)
-      configure_sssd_service(config)
-      configure_sssd_pam(config)
-      configure_sssd_ifp(config)
-      debug_msg("- Creating #{SSSD_CONFIG}")
-      config_file_write(config, SSSD_CONFIG, timestamp)
+  class Sssd < Base
+    attr_accessor :sssd
+    attr_accessor :opts
+
+    def initialize(opts = {})
+      @opts = opts
+      @sssd = nil
     end
 
-    private
+    def load(file_path)
+      @sssd = IniParse.open(file_path)
+    end
 
-    def configure_sssd_domain(config, domain)
-      ldap_user_extra_attrs = LDAP_ATTRS.keys.join(", ")
-      if config.include?("ldap_user_extra_attrs = ")
-        pattern = "[domain/#{Regexp.escape(domain)}](\n.*)+ldap_user_extra_attrs = (.*)"
-        config[/#{pattern}/, 2] = ldap_user_extra_attrs
-      else
-        pattern = "[domain/#{Regexp.escape(domain)}].*(\n)"
-        config[/#{pattern}/, 1] = "\nldap_user_extra_attrs = #{ldap_user_extra_attrs}\n"
-      end
+    def save(file_path)
+      return unless sssd
+      config_file_backup(file_path)
+      info_msg("Saving SSSD to #{file_path}")
+      sssd.save(file_path)
+    end
 
-      pattern = "[domain/#{Regexp.escape(domain)}].*(\n)"
-      config[/#{pattern}/, 1] = "\nentry_cache_timeout = 600\n"
+    def configure_domain(domain)
+      domain = section("domain/#{domain}")
+      domain["ldap_user_extra_attrs"] = LDAP_ATTRS.keys.join(", ")
+      domain["entry_cache_timeout"] = 600
     end
 
-    def configure_sssd_service(config)
-      services = config.match(/\[sssd\](\n.*)+services = (.*)/)[2]
-      services = "#{services}, ifp" unless services.include?("ifp")
-      config[/\[sssd\](\n.*)+services = (.*)/, 2] = services
+    def add_service(service)
+      services = section("sssd")["services"]
+      services = (services.split(",").map(&:strip) | [service]).join(", ")
+      sssd.section("sssd")["services"] = services
+      sssd.section(service)
     end
 
-    def configure_sssd_pam(config)
-      return if config.include?("[pam]")
-      config << "\n[pam]\n"
+    def configure_ifp
+      add_service("ifp")
+      ifp = section("ifp")
+      ifp["allowed_uids"] = "#{APACHE_USER}, root"
+      ifp["user_attributes"] = LDAP_ATTRS.keys.collect { |k| "+#{k}" }.join(", ")
     end
 
-    def configure_sssd_ifp(config)
-      user_attributes = LDAP_ATTRS.keys.collect { |k| "+#{k}" }.join(", ")
-      ifp_config      = "
-allowed_uids = #{APACHE_USER}, root
-user_attributes = #{user_attributes}
-"
-      if config.include?("[ifp]")
-        if config[/\[ifp\](\n.*)+user_attributes = (.*)/]
-          config[/\[ifp\](\n.*)+user_attributes = (.*)/, 2] = user_attributes
-        else
-          config[/\[ifp\](\n)/, 1] = ifp_config
-        end
-      else
-        config << "\n[ifp]#{ifp_config}\n"
+    private
+
+    def section(key)
+      if key == "domain"
```

Comment:
@abellotti I think we can eliminate these 2 checks and only user
`if key =~ /^domain\/.* $/`